### ISSUE TITLE
ReportUnnecessaryComparison bogus error 2

### ISSUE THREAD

**Issue by rafrafek** [microsoft/pyright#7838]
Hi, I'm getting a similar error to #7431 using the latest VS Code and Pylance and also on `pyright 1.1.361` running from a terminal.

Code:

```python
def main() -> None:
    counter = 0

    def foo() -> None:
        nonlocal counter
        if counter == 2:
            print(counter)
        counter += 1

    foo()
    foo()
    foo()


main()
```

Error:

```
Condition will always evaluate to False since the types "Literal[0]" and "Literal[2]" have no overlap
Pylance reportUnnecessaryComparison
```

Screenshot:

<img width="864" alt="image" src="https://github.com/microsoft/pyright/assets/23004737/f4e3c29e-cd57-4f46-8d74-1b61f8f0cd80">

_Originally posted by @rafrafek in https://github.com/microsoft/pyright/issues/7431#issuecomment-2039742433_

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.362, which I just published.
